Makefile

```diff
@@ -160,3 +128,88 @@ golangci-lint-fix.%: $(GOLANGCI_LINT)
 
 .PHONY: FORCE # use this to force phony behavior for targets with pattern rules
 FORCE:
+
+##@ Deployment
+
+.PHONY: cluster
+cluster: kind ctlptl ## Create Kind cluster and local registry
+	$(CTLPTL) apply -f ctlptl.yaml
+
+.PHONY: cluster-reset
+cluster-reset: kind ctlptl ## Delete Kind cluster
+	$(CTLPTL) delete -f ctlptl.yaml
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: deploy
+deploy: .gen kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build . | $(KUBECTL) apply -f -
+
+.PHONY: undeploy
+undeploy: kustomize ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build . | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
+
+##@ Tools
+
+## Location to install dependencies to
+TOOLBIN ?= $(CURDIR)/.cache/tools
+$(TOOLBIN):
+	mkdir -p $(TOOLBIN)
```

Comment:
I have found that `go mod vendor` seems to be a simpler way of downloading tools to a local cache. I used that method with the client generator and found it worked well. Then we can get rid of `go-install-tool` entirely for something like

```
tools:
  cd hack/tools && go mod vendor -o $(TOOLBIN)
```

WDYT?

Reply:
I don't like the fact that `go run <tool>` does not cache, and always rebuilds the binary. While not an issue in CI, this is not the best for local development.